service_container/3.3-di-changes.rst

```diff
@@ -0,0 +1,740 @@
+
+This does *not* work for all tags. Many tags have *required* attributes, like event
+*listeners*, where you also need to specify the event name and method in your tag.
+Autoconfigure works only for tags without any required tag attributes, and as you
+read the docs for a feature, it'll tell you whether or not the tag is needed. You
+can also look at the extension classes (e.g. `FrameworkExtension for 3.3.0`_) to
+see what it autoconfigures.
+
+    What if I need to add a priority to my tag?
+
+Many autoconfigured tags have an optional priority. If you need to specify a priority
+(or any other optional tag attribute), no problem! Just :ref:`manually configure your service <services-manually-wire-args>`
+and add the tag. Your tag will take precedent over the one added by auto-configuration.
+
+What about Performance
+----------------------
+
+Symfony is unique because it has a *compiled* container. This means that there is
+*no* runtime performance impact for using any of these features. That's also why
+the autowiring system can give you such clear errors.
+
+However, there is some performance impact in the ``dev`` environment. Most importantly,
+your container will likely be rebuilt more often when you modify your service classes.
+This is because it needs to rebuild whenever you add a new argument to a service,
+or add an interface to your class that should be autoconfigured.
+
+In very big projects, this may be a problem. If it is, you can always opt to *not*
+use autowiring. If you think the cache rebuilding system could be smarter in some
+situation, please open an issue!
+
+Upgrading to the new Symfony 3.3 Configuration
+----------------------------------------------
+
+Ready to upgrade your existing project? Great! Suppose you have the following configuration:
+
+.. code-block:: yaml
+
+    # app/config/services.yml
+    services:
+        app.github_notifier:
+            class: AppBundle\Service\GitHubNotifier
+            arguments:
+                - '@app.api_client_github'
+
+        markdown_transformer:
+            class: AppBundle\Service\MarkdownTransformer
+
+        app.api_client_github:
+            class: AppBundle\Service\ApiClient
+            arguments:
+                - 'https://api.github.com'
+
+        app.api_client_sl_connect:
+            class: AppBundle\Service\ApiClient
+            arguments:
+                - 'https://connect.sensiolabs.com/api'
+
+It's optional, but let's upgrade this to the new Symfony 3.3 configuration step-by-step,
+*without* breaking our application.
+
+Step 1): Adding _defaults
+~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Start by adding a ``_defaults`` section with ``autowire`` and ``autoconfigure``.
+
+.. code-block:: diff
+
+    # app/config/services.yml
+    services:
+    +    _defaults:
+    +        autowire: true
+    +        autoconfigure: true
+
+        # ...
+
+This step is easy: you're already *explicitly* configuring all of your services.
```

Comment:
this remind me of `@required`, we need to tell people about it also (maybe not in this PR)